lib/matplotlib/figure.py

```diff
@@ -213,435 +219,139 @@ def update(self, left=None, bottom=None, right=None, top=None,
-        """
+    def __init__(self):
         super().__init__()
         # remove the non-figure artist _axes property
         # as it makes no sense for a figure to be _in_ an axes
         # this is used by the property methods in the artist base class
         # which are over-ridden in this class
         del self._axes
-        self.callbacks = cbook.CallbackRegistry()
-
-        if figsize is None:
-            figsize = mpl.rcParams['figure.figsize']
-        if dpi is None:
-            dpi = mpl.rcParams['figure.dpi']
-        if facecolor is None:
-            facecolor = mpl.rcParams['figure.facecolor']
-        if edgecolor is None:
-            edgecolor = mpl.rcParams['figure.edgecolor']
-        if frameon is None:
-            frameon = mpl.rcParams['figure.frameon']
-
-        if not np.isfinite(figsize).all() or (np.array(figsize) < 0).any():
-            raise ValueError('figure size must be positive finite not '
-                             f'{figsize}')
-        self.bbox_inches = Bbox.from_bounds(0, 0, *figsize)
-
-        self.dpi_scale_trans = Affine2D().scale(dpi)
-        # do not use property as it will trigger
-        self._dpi = dpi
-        self.bbox = TransformedBbox(self.bbox_inches, self.dpi_scale_trans)
-
-        self.transFigure = BboxTransformTo(self.bbox)
-
-        self.patch = Rectangle(
-            xy=(0, 0), width=1, height=1, visible=frameon,
-            facecolor=facecolor, edgecolor=edgecolor, linewidth=linewidth,
-            # Don't let the figure patch influence bbox calculation.
-            in_layout=False)
-        self._set_artist_props(self.patch)
-        self.patch.set_antialiased(False)
 
-        FigureCanvasBase(self)  # Set self.canvas.
         self._suptitle = None
 
-        if subplotpars is None:
-            subplotpars = SubplotParams()
-
-        self.subplotpars = subplotpars
-
         # constrained_layout:
         self._layoutgrid = None
-        self._constrained = False
-
-        self.set_tight_layout(tight_layout)
-
-        self._axstack = _AxesStack()  # track all figure axes and current axes
-        self.clf()
-        self._cachedRenderer = None
-
-        self.set_constrained_layout(constrained_layout)
-        # stub for subpanels:
-        self.panels = []
-        self.transPanel = self.transFigure
 
         # groupers to keep track of x and y labels we want to align.
         # see self.align_xlabels and self.align_ylabels and
         # axis._get_tick_boxes_siblings
         self._align_xlabel_grp = cbook.Grouper()
         self._align_ylabel_grp = cbook.Grouper()
 
+        self.figure = self
         # list of child gridspecs for this figure
         self._gridspecs = []
+        self._localaxes = _AxesStack()  # keep track of axes at this level
+        self.artists = []
+        self.lines = []
+        self.patches = []
+        self.texts = []
+        self.images = []
+        self.legends = []
+        self.panels = []
```

Comment:
Should the attribute still be `panels`? It can't be `subfigures` of course, because of the method, but then there's not really any documented relation here. Or perhaps this should be private.